portal-api/src/v1/users/validation/index.js

```diff
@@ -32,21 +34,29 @@ const updateWithoutCurrentPasswordRules = [
   readOnlyRoleCannotBeAssignedWithOtherRoles,
   usernameAndEmailMustMatch,
   emailMustBeValidEmailAddress,
+  emailMustBeUnique,
 ];
 
 const updateWithCurrentPasswordRules = [...updateWithoutCurrentPasswordRules, currentPasswordMustMatch];
 
-const applyRules = (ruleset, existingUser, candidateChange) =>
-  ruleset.reduce((accumulator, rule) => {
-    const result = rule(existingUser, candidateChange);
-    return result.length ? accumulator.concat(result) : accumulator;
-  }, []);
+const applyRules = async (ruleset, existingUser, candidateChange) => {
+  let errors = [];
+  for (const rule of ruleset) {
+    const result = await rule(existingUser, candidateChange);
 
-const applyCreateRules = (candidateChange) => applyRules(createRules, null, candidateChange);
+    if (result.length) {
+      errors = errors.concat(result);
+    } // We concat here as rules return an array
+  }
 
-const applyUpdateRules = (existingUser, candidateChange) => {
-  const rule = !candidateChange.currentPassword ? updateWithoutCurrentPasswordRules : updateWithCurrentPasswordRules;
-  return applyRules(rule, existingUser, candidateChange);
+  return errors;
+};
+
+const applyCreateRules = async (candidateChange) => await applyRules(createRules, null, candidateChange);
+
```

Comment:
Could we add some documentation? Maybe tech debt since this file is lacking documentation